src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/UnixPkcs12Reader.cs

```diff
@@ -115,22 +124,24 @@ public void Dispose()
             // Generally, having a MemoryManager cleaned up in a Dispose is a bad practice.
             // In this case, the UnixPkcs12Reader is only ever created in a using statement,
             // never accessed by a second thread, and there isn't a manual call to Dispose
-            // mixed in anywhere.
-            if (_tmpManager != null)
+            // mixed in anywhere outside of an aborted allocation path.
+
+            PointerMemoryManager<byte>? manager = Interlocked.Exchange(ref _tmpManager, null);
```

Comment:
```suggestion
            PointerMemoryManager<byte>? manager = _tmpManager;
            _tmpManager = null;
```
`Interlocked.Exchange` makes it look like that this type is supposed to be thread safe, that it is not.

Reply:
Probably just because it's a shorter line of code, and a million years ago Vance said Interlocked was basically free so long as there's not actually contention.

I've gone ahead and converted them all to not use it.